src/osd/PG.cc

```diff
@@ -4621,10 +4656,23 @@ void PG::chunky_scrub(ThreadPool::TPHandle &handle)
         assert(last_update_applied >= scrubber.subset_last_update);
 
         // build my own scrub map
-        ret = build_scrub_map_chunk(scrubber.primary_scrubmap,
-                                    scrubber.start, scrubber.end,
-                                    scrubber.deep, scrubber.seed,
-				    handle);
+	{
+	  epoch_t epoch = get_osdmap()->get_epoch();
+	  if (scrubber.preempt_left > 0) {
+	    unlock();
```

Comment:
@liewegas . I think we should remove the condition "scrubber.preempt_left > 0". preempt mean write-object in [scrubber.start, scrubber.end]. If preempt_left= 0, write-object in [start,end] blocked by write_blocked_by_scrub. write-object not in this range like the "preempt_left > 0 " can update.  Or I missing something?